File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(n):
    e = 0
    for x in range(1,n+1):
        e += x
    return e*e


def sum_of_squares(n):
    e = 0
    for x in range(1,n+1):
        e += x*x
    return e


def difference_of_squares(n):
    return square_of_sum(n)-sum_of_squares(n)

File: solutions/javascript/difference-of-squares/1/difference-of-squares.js
//
// This is only a SKELETON file for the 'Difference Of Squares' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Squares {
  constructor(x) {
    this.x = x;
    this.sum;
  }

  get sumOfSquares() {
    let sum = 0;
    for (let y = 1; y <= this.x; y++) {
      sum += y ** 2;
    }
    return sum;
  }

  get squareOfSum() {
    let sum = 0;
    for (let y = 1; y <= this.x; y++) {
      sum += y;
    }
    return sum ** 2;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}
